src/main/java/org/sagebionetworks/bridge/exporter/worker/ExportWorkerManager.java

```diff
@@ -627,4 +644,34 @@ public void endOfStream(ExportTask task) throws RestartBridgeExporterException {
     static boolean isSynapseDown(Throwable t) {
         return (t instanceof SynapseServerException && ((SynapseServerException)t).getStatusCode() == 503);
     }
+
+    // For redrives, we need to know whether an exception is retryable or not. If it is, we can redrive it. If not, we
+    // log the exception but otherwise swallow it.
+    //
+    // Package-scoped for unit tests.
+    static boolean isRetryable(Throwable t) {
+        if (t == null) {
+            // This should never happen. But if it does, assume something really bad happened and don't try to redrive.
+            return false;
+        } else if (t instanceof BridgeSDKException) {
+            // If this is a Bridge exception in the 400s, this is not retryable.
+            int statusCode = ((BridgeSDKException) t).getStatusCode();
+            if (statusCode >= 400 && statusCode < 500) {
+                return false;
+            }
+        } else if (t instanceof JsonProcessingException || t instanceof JsonParseException ||
+                t instanceof MalformedJsonException) {
+            // JSON parse exceptions are generally also not retryable. Note that BridgeEX uses Jackson, but
+            // BridgeJavaSDK uses GSON, so we need to handle both.
+            // Jackson uses JsonProcessingException as their base exception.
+            // GSON uses JsonParseException and MalformedJsonException.
+            return false;
```

Comment:
Should we file a JIRA to hide dependent library exceptions like these and rethrow them as BridgeSDKExceptions? It seems unfortunate you have to know about the rest-client's serialization implementation. We'd have to figure out if we can intercept this in the generated code, I think.